### PR TITLE
feat(product): adds daff prefix to exported members of product

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, we don't prefix our product module properly.

## What is the new behavior?
This adds prefixes.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

Any depend packages must use the new exported type with the prefix.

## Other information

### CODE REVIEW

File: libs/product/src/reducers/product-entities.reducer.ts [hunk 1]
```diff
@@ -1,33 +1,33 @@
 import { createEntityAdapter, EntityState, EntityAdapter, Dictionary } from '@ngrx/entity';
 
-import { ProductGridActionTypes, ProductGridActions } from '../actions/product-grid.actions';
-import { ProductActionTypes, ProductActions } from '../actions/product.actions';
```
Comment (lderrickable) [inline libs/product/src/reducers/product-entities.reducer.ts]: DaffProductActions?

**Comment by lderrickable** [comment]
I think `StateProductModule` should be `DaffStateProductModule`, right?

**Comment by lderrickable** [comment]
I think `StateProductModule` should be `DaffStateProductModule`, right?

File: libs/product/src/reducers/product-entities.reducer.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/product/src/reducers/product-entities.reducer.ts]: Addressed in 12b6a9f55d655fd6d185281b323c6d341f1f9a9e

**Pull request merged**